### DISCUSSION TITLE
Can we use ajax on text field?

### DISCUSSION THREAD

**Discussion by realtebo** [Laravel-Backpack/community-forum#646] [Q&A (Help)] [answered]
I am creating an "a bit complex" form.

After a lot of using of new `CrudJs `Library, I ended with this not so strange use case

After user change a select, we need to prepopulate a text field with a value based both on other form fields AND db.

The ideal, in this case, should be to have `'ajax' => true ` and` 'deprends' => [<list of other fields>]`, even for `CRUD::field` of type `text`.
Se our server would receive all form data, can access db, and returns to backpack the data to show

But this is not the way it works now... so ... 

any ideas?

**Comment by pxpm** [comment]
The current implementation as you say does not contemplate ajax for text fields. 

You can write a simple script inside your `onChange()` events (no need to overwrite the field), serialize the form and send everything to your endpoint. wait for the response and populate the field. 

```javascript
let form = element.closest('form');

fetch(https://your-endpoint, {form: form.serializeArray()})
  .then(data => {
    console.log('Response data:', data);
  })
  .catch(error => {
    console.error('Fetch error:', error);
  });
```

Cheers

**Reply by realtebo** [reply]
Yes, thanks. We know how to manually make an ajax call. 
It was just a suggestion to extend these powerful features to the text field (ajax and depends)

**Reply by pxpm** [reply, marked as answer]
I think if we ever implement something like ajax for all the fields it would be using some other technology like livewire or similar. And it will bring full ajax forms functionality including validation etc. 

Implementing this in all fields now just for the "dependency" would be a huge change without the full benefits of an ajax solution. 

I am not sure if/how/when we are going to do it, I would like to have it for sure, but for now we need to stick to a bit more work to create those special use cases, sorry 🙏

**Reply by realtebo** [reply]
I understand
